Some random fixes

lkundrak opened the pull request:
Here's a fairly random collection of changes I chose to make when making Lumail work for me.

The only real fix there is the removal of the 1 second wait on spawning the IMAP helper, since that is too short for some of my machines (oh the lovely OLPC laptop); while perhaps being unnecessarily long for others. It probably deserves a better fix anyway though; but this seems to work well for now.

skx commented on src/imap_proxy.h:
```diff
@@ -64,4 +64,5 @@ class CIMAPProxy : public Singleton<CIMAPProxy>
      * The handle to our child-process.
      */
     pid_t m_child;
+    std::string sock_path;
```
Please add a comment here :)

skx commented:
I'm glad you were able to make such an impressive pull-request, I'll hope that means the code was clean and easy to understand!

If you fixup the member-name, and comment, then I'll merge happily.

Thank-you for your interest and contribution :)

lkundrak commented:
> I'm glad you were able to make such an impressive pull-request, I'll hope that means the code was clean and easy to understand!

Well, if this is impressive then I must say you have pretty low standards :)

But yes, the code I've seen is reasonably easy to follow.

> If you fixup the member-name, and comment, then I'll merge happily.

I've now pushed an updated branch that addresses your review.

> Thank-you for your interest and contribution :)

Well, thanks for making lumail free software in the first place.

skx commented:
Impressive mostly because it touched both sides of the c++/perl boundary.  Anyway merging now, thanks-again !
